pipeline/metadata/flatten_satellite.py

```diff
@@ -565,3 +502,63 @@ def _process_satellite_v2_control(
         row['received'] = all_received
 
         yield row
+
+
+class FlattenBlockpages(beam.DoFn):
+  """DoFn class for flattening lines of blockpage text into Rows."""
+
+  def setup(self) -> None:
+    #pylint: disable=attribute-defined-outside-init
+    self.blockpage_matcher = BlockpageMatcher()
+    #pylint: enable=attribute-defined-outside-init
+
+  def process(self, element: Tuple[str, str]) -> Iterator[BlockpageRow]:
+    (filename, line) = element
+
+    try:
+      scan = json.loads(line)
+    except json.decoder.JSONDecodeError as e:
+      logging.warning('JSONDecodeError: %s\nFilename: %s\n%s\n', e, filename,
+                      line)
+      return
+
+    yield from self._process_satellite_blockpages(scan, filename)
+
+  def _process_satellite_blockpages(self, blockpage_entry: ResponsesEntry,
+                                    filepath: str) -> Iterator[BlockpageRow]:
+    """Process a line of Satellite blockpage data.
+
+    Args:
+      blockpage_entry: a loaded json object containing the parsed content of the line
+      filepath: a filepath string like "<path>/blockpages.json.gz"
+
+    Yields:
+      BlockpageRows, usually 2 corresponding to the fetched http and https data
+    """
+    row = {
+        'domain': blockpage_entry['keyword'],
+        'ip': blockpage_entry['ip'],
+        'date': blockpage_entry['start_time'][:10],
+        'start_time': format_timestamp(blockpage_entry['start_time']),
+        'end_time': format_timestamp(blockpage_entry['end_time']),
+        'success': blockpage_entry['fetched'],
+        'source': flatten_base.source_from_filename(filepath),
+    }
+
+    http = {
+        'https': False,
+    }
+    http.update(row)
+    received_fields = flatten_base.parse_received_data(
+        self.blockpage_matcher, blockpage_entry.get('http', ''), True)
+    http.update(received_fields)
+    yield http
```

Comment:
Why do you need to emit the http and the https results separately? In the end you will be joining both into the same measurement again, no?

Reply:
They're split because they're fundamentally two independent measurements. We'll probably end up joining both but you could imagine wanting to do different things with them. (Join only the http, or only https, or just look at them independently of the dns.) This way they can be similar to our existing http/s measurement schema, instead of having to have two versions of all their shared fields like `http_received_headers` + `https_received_headers`

Reply:
This is fine with me, but you are essentially creating two new types that are unnecessary (one for HTTP, one for HTTPS).

I believe the pipeline would be more efficient and simpler if the entries from the metadata files were available in their original formats in a big join, so you can extract what you need to create the final output without creating intermediate types. (that's the suggestion in the doc I shared)